selfservice/strategy/oidc/strategy_settings.go

```diff
@@ -390,7 +396,21 @@ func (s *Strategy) initLinkProvider(w http.ResponseWriter, r *http.Request, ctxU
 		return err
 	}
 
-	codeURL, err := getAuthRedirectURL(ctx, provider, req, state, up)
+	if ok, err := MaybeUsePKCE(ctx, s.d, provider, ctxUpdate.Flow); err != nil {
+		return s.handleError(w, r, ctxUpdate.Flow, p.Link, nil, errors.WithStack(herodot.ErrInternalServerError.WithReason("Could not update flow context").WithDebug(err.Error())))
+	} else {
+		span.SetAttributes(attribute.Bool("pkce", ok))
+	}
+
+	if err := storeProvider(ctxUpdate.Flow, p.Link); err != nil {
+		return s.handleError(w, r, ctxUpdate.Flow, p.Link, nil, errors.WithStack(herodot.ErrInternalServerError.WithReason("Could not update flow context").WithDebug(err.Error())))
+	}
+
+	if err := s.d.SettingsFlowPersister().UpdateSettingsFlow(ctx, ctxUpdate.Flow); err != nil {
+		return s.handleError(w, r, ctxUpdate.Flow, p.Link, nil, errors.WithStack(herodot.ErrInternalServerError.WithReason("Could not update flow").WithDebug(err.Error())))
+	}
+
+	codeURL, err := getAuthRedirectURL(ctx, provider, ctxUpdate.Flow, state, up)
```

Comment:
`ctxUpdate.Flow` does not represent the most up-to-date version of the flow. Use the return value of `_, err = s.validateFlow(ctx, r, ctxUpdate.Flow.ID)` instead (like it was before)

Reply:
done